lib/THC/THCReduceApplyUtils.cuh

```diff
@@ -19,64 +19,101 @@ __device__ __forceinline__ IndexType getLinearBlockId() {
     blockIdx.x;
 }
 
-// Block-wide reduction in shared memory helper; only threadIdx.x == 0 will
-// return the reduced value
-template <typename T, typename ReduceOp>
-__device__ T reduceBlock(T* smem,
-                         int numVals,
-                         T threadVal,
-                         ReduceOp reduceOp,
-                         T init) {
+// Return value is in threadVals for threadIdx.x == 0
+template <typename T, typename ReduceOp, int N>
+__device__ void reduceNValuesInBlock(T *smem,
```

Comment:
add a comment above that this helper is meant to reduce N different values concurrently, or something